### ISSUE TITLE
open /dev/disk/by-uuid: no such file or directory

### ISSUE THREAD

**Issue by tkellen** [kubernetes-incubator/cri-containerd#509]
Hello,

First, thanks for all your great work on this project!

I am attempting to run cri-containerd v1.0.0-beta.0 on [Scaleway](http://scaleway.com) Ubuntu machines and I'm hitting this:

```root@scaleway-node-1:/etc/systemd/system# /usr/local/bin/cri-containerd --logtostderr
I1218 17:54:40.732642    5698 cri_containerd.go:100] Run cri-containerd &{Config:{ContainerdConfig:{RootDir:/var/lib/containerd Snapshotter:overlayfs Endpoint:/run/containerd/containerd.sock Runtime:io.containerd.runtime.v1.linux RuntimeEngine: RuntimeRoot:} CniConfig:{NetworkPluginBinDir:/opt/cni/bin NetworkPluginConfDir:/etc/cni/net.d} SocketPath:/var/run/cri-containerd.sock RootDir:/var/lib/cri-containerd StreamServerAddress: StreamServerPort:10010 CgroupPath: EnableSelinux:false SandboxImage:gcr.io/google_containers/pause:3.0 StatsCollectPeriod:10 SystemdCgroup:false OOMScore:-999 EnableProfiling:true ProfilingPort:10011 ProfilingAddress:127.0.0.1} ConfigFilePath:/etc/cri-containerd/config.toml}
Error: failed to create CRI containerd service: failed to get imagefs uuid of "/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs": open /dev/disk/by-uuid: no such file or directory
```

...any insight on how to proceed would be greatly appreciated!

**Comment by Random-Liu** [comment]
Are you using `udev` or `mdev`?

**Comment by Random-Liu** [comment]
I filed https://github.com/kubernetes-incubator/cri-containerd/pull/510 for now to unblock people from trying it out.

I'll work with @yujuhong and the other team members to figure out whether we should change CRI to make this more portable.

**Comment by Random-Liu** [comment]
@tkellen Thanks for filing the issue. I've filed https://github.com/kubernetes/kubernetes/issues/57356 to carry on the related api design discussion.

**Comment by tkellen** [comment]
Thanks @Random-Liu! Apologies for not replying RE: mdev/udev, yet. This was my first time using Scaleway and I shut the instances down shortly after the failure and moved on to the next provider (I'm testing any with terraform support over at https://github.com/tkellen/kwm/tree/master/hack.

I'll spin up another machine tomorrow and check. If you give me some commands to run I'd be happy to share the output.

**Comment by Random-Liu** [comment]
@tkellen Thanks! It seems that you can check `/sbin/mdev`, see https://stackoverflow.com/questions/22719005/how-to-know-if-udev-or-mdev-is-enabled.

**Comment by tkellen** [comment]
```
root@scaleway-controlplane-0:~# echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
root@scaleway-controlplane-0:~# which mdev
root@scaleway-controlplane-0:~# cat /proc/sys/kernel/hotplug
/sbin/hotplug
root@scaleway-controlplane-0:~# which hotplug
root@scaleway-controlplane-0:~#
```

**Comment by tkellen** [comment]
Seems neither are available. Picking a random person from the scaleway organization to see if they can weigh in on this decision. Hi... @nlm?

**Comment by Random-Liu** [comment]
@tkellen Thanks for investigating that.
@nlm, what device manager is scaleway Ubuntu using? Why isn't there `/dev/disk/by-uuid`?

**Comment by Random-Liu** [comment]
@tkellen We added `skip-imagefs-uuid` flag, and will be included in `v1.0.0-beta.1`. If the current uuid based approach doesn't work, user could use that flag to skip that step. It should not prevent user from trying out this.

I'll close this issue. Further discussion will be carried on in the kubernetes issue kubernetes/kubernetes#57356.

**Comment by tkellen** [comment]
Thanks @Random-Liu!
